src/masoniteorm/connections/decorators.py.49ec9b0f8ff47a3ba27e2805c82b56ea.tmp

```diff
@@ -0,0 +1,20 @@
+from config.database import db
+
+
+class Transaction(object):
+
+    def __init__(self, connection="default"):
+        self.connection = connection
+
+    def __call__(self, fn, *args, **kwargs):
+        # before function
+        db.begin_transaction(self.connection)
+        try:
+            result = fn(*args, **kwargs)
+        except:
+            db.rollback(self.connection)
+            return None
+
+        # after function
+        db.commit(self.connection)
+        return result
```

Comment:
The reason you are getting the weird user error is because decorator are a. pain in the ass to work with. Look closely what you're doing here. 

Because the result contains the user class, you are returning the user class here.